tools/run_examples.py

```diff
@@ -0,0 +1,190 @@
+import argparse
+import os
+from pathlib import Path
+import subprocess
+import sys
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+EXAMPLES_DIR = Path(__file__).parents[1] / Path("examples")
+ALL_EXAMPLES: List[str] = [str(p) for p in EXAMPLES_DIR.glob("**/*.py")]
+# TODO (crcrpar): Allow fastaiv2 example once https://forums.fast.ai/t/ganlearner-error-no-implementation-found-on-types-that-implement-invisibletensor/83451/7 gets resolved.  # NOQA
+_DEFAULT_DENY_LIST: List[str] = [
+    "fastai",
+    "hydra/simple",
+    "pytorch_distributed_",
+    "rapids_",
+    "allennlp/subsample_dataset_reader",
+]
+VER_TO_DENY_LIST: Dict[Tuple[int, int], List[str]] = {
+    (3, 6): _DEFAULT_DENY_LIST + ["botorch_"],
+    (3, 7): _DEFAULT_DENY_LIST,
+    (3, 8): _DEFAULT_DENY_LIST + ["keras_", "tensorboard_", "tensorflow_", "tfkeras_"],
+}
+
+
+def _get_distributed_examples(files: List[str]) -> List[str]:
+    return [f for f in files if "distributed" in f]
+
+
+def _get_ipynb_examples(files: Optional[List[str]] = None) -> List[str]:
+    if files is None:
+        return [str(p) for p in EXAMPLES_DIR.glob("**/*.ipynb")]
+    else:
+        return [f for f in files if "ipynb" in f]
+
+
+def _get_multinode_examples(files: List[str]) -> List[str]:
+    return [f for f in files if "chainermn" in f]
+
+
+def _get_hydra_examples(files: List[str]) -> List[str]:
+    return [f for f in files if "hydra" in f]
+
+
+class ExampleFiles:
+    def __init__(self, files: Optional[List[str]] = None) -> None:
+        _all_examples = ALL_EXAMPLES
+        deny_list = VER_TO_DENY_LIST[sys.version_info[:2]]
+        if files is not None:
+            _all_examples = [f for f in files if f in _all_examples]
+
+        all_examples = []
+        for f in _all_examples:
+            if not any(d in f for d in deny_list):
+                all_examples.append(f)
+
+        self.distributed_examples = _get_distributed_examples(all_examples)
+        self.ipynb_examples = _get_ipynb_examples(all_examples)
+        self.multinode_examples = _get_multinode_examples(all_examples)
+        self.hydra_examples = _get_hydra_examples(all_examples)
+        self.vanilla_examples = list(
+            set(all_examples)
+            - set(self.distributed_examples)
+            - set(self.multinode_examples)
+            - set(self.hydra_examples)
+        )
+
+
+def parse_args():
```

Comment:
Not a strong opinion, but the type hinting is missing, so how about adding for consistency? 
```suggestion
def parse_args() -> argparse.NameSpace:
```

The same is true for the other functions.